### PR TITLE
Method annotation is not enforcing return type argument

### PR DESCRIPTION
**Pull request by gmponos**
Since the class is not final and can be extended the method annotation does not enforce the return type.

### CODE REVIEW

File: tests/Exception/BadResponseExceptionTest.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        $prev = new \Exception();
+        $response = new Response();
+        $e = new BadResponseException('foo', $req, $response, $prev);
+        self::assertSame($req, $e->getRequest());
+        self::assertSame($response, $e->getResponse());
+        self::assertTrue($e->hasResponse());
+        self::assertSame('foo', $e->getMessage());
+        self::assertSame($prev, $e->getPrevious());
+    }
+}
```
Comment (sagikazarmark) [inline tests/Exception/BadResponseExceptionTest.php]: Missing newline

Comment (gmponos) [reply tests/Exception/BadResponseExceptionTest.php]: fixed..

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -200,6 +200,11 @@ parameters:
 			count: 1
 			path: src/Exception/BadResponseException.php
 
+		-
+			message: "#^Method GuzzleHttp\\\\Exception\\\\BadResponseException\\:\\:getResponse\\(\\) should return Psr\\\\Http\\\\Message\\\\ResponseInterface but returns Psr\\\\Http\\\\Message\\\\ResponseInterface\\|null\\.$#"
+			count: 1
+			path: src/Exception/BadResponseException.php
+
```
Comment (gmponos) [inline phpstan-baseline.neon]: I added this to baseline

IMHO it is a false positive.

Comment (sagikazarmark) [reply phpstan-baseline.neon]: Might even be a bug

Comment (gmponos) [reply phpstan-baseline.neon]: No it is not... phpstan only looks at the docblock..  and does not check the whole flow of the code in order to understand that the response can never be null...

Comment (sagikazarmark) [reply phpstan-baseline.neon]: What if you add a docblock to the function then?

**Pull request merged**